src/os/FileStore.cc

```diff
@@ -3780,6 +3780,7 @@ int FileStore::_fgetattrs(int fd, map<string,bufferptr>& aset)
     dout(10) << " -ERANGE, got " << len << dendl;
     if (len < 0) {
       assert(!m_filestore_fail_eio || len != -EIO);
+      delete[] names2;
```

Comment:
May be it's better to use std::unique_ptr to avoid the same issue if code changes?